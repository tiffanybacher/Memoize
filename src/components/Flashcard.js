import React, { Component } from 'react';

class Flashcard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    if (this.props.displayAnswer) {
      var answer = 
        <p className="Flashcard-answer">
          {this.props.answer}
        </p>
    }

    return (
      <article className="Flashcard">
        <h3 className="Flashcard-question">
          {this.props.question}
        </h3>
        {answer}
      </article>
    )
  }
}

export default Flashcard;
