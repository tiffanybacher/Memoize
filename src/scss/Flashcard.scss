.Flashcard {
  @include border();
  background-color: white;
  // height: 200px;
  margin: 0 auto 30px;

  .Flashcard-question {
    margin: 30px 0;
    padding: 0 20px;
    text-align: center;
  }

  .Flashcard-answer {
    font-size: 1.2em;
    margin: 0;
    padding: 0 30px 30px;
  }

  .show-btn {
    background-color: $purple;
    border-top: 3px solid $purple;
    border-right: 3px solid $purple;
    border-bottom: none;
    border-left: none;
    border-radius: 0 3px 0 0;
    color: white;
    font-family: "Montserrat";
    font-size: 1em;
    font-weight: 600;
    padding: 10px;
    margin: 0;
  }
}

// iPad
@media only screen and (min-width: 768px) {
  
  .Flashcard {
    // height: 300px;
  }
}

// iPad Pro
@media only screen and (min-width: 1024px) {

  .Flashcard {
    // height: 400px;
  }
}