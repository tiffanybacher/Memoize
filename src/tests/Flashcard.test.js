import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Flashcard from '../components/Flashcard'

const mockData = {
    "question": "What is an object in JavaScript?",
    "answer": "An object in JavaScript can be compared to an object in real life. It is a stand alone entity with it's own set of properties and methods.",
    "id": 1,
    "group": 1
  }
let mockAnswerIsShown = false;

describe('Flashcard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Flashcard 
        question={mockData.question}
        answer={mockData.answer}
        answerIsShown={mockAnswerIsShown} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if answerIsShown is true', () => {
  mockAnswerIsShown = true;

  expect(wrapper).toMatchSnapshot();
  });
});