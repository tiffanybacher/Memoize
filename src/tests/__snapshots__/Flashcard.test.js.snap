// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flashcard should match snapshot 1`] = `
<article
  className="Flashcard"
>
  <h3
    aria-live="polite"
    className="Flashcard-question"
  >
    What is an object in JavaScript?
  </h3>
</article>
`;

exports[`Flashcard should match snapshot if answerIsShown is true 1`] = `
<article
  className="Flashcard"
>
  <h3
    aria-live="polite"
    className="Flashcard-question"
  >
    What is an object in JavaScript?
  </h3>
</article>
`;
